src/collective/collectionfilter/validator.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+from Acquisition import aq_base, aq_inner
+from plone.dexterity.interfaces import IDexterityContent
+from z3c.form import validator
+from z3c.form.interfaces import IValidator
+from zope.component import adapter, getMultiAdapter, queryUtility
+from zope.interface import Interface, Invalid, implementer
+from zope.schema.interfaces import IField
+
+from collective.collectionfilter import _
+from collective.collectionfilter.interfaces import (
+    ICollectionFilterBaseSchema, ICollectionFilterBrowserLayer)
+
+
+@implementer(IValidator)
+@adapter(Interface, ICollectionFilterBrowserLayer, Interface, IField, Interface)
+class TargetCollectionValidator(validator.SimpleFieldValidator):
+
+    def validate(self, value):
+        super(TargetCollectionValidator, self).validate(value)
+
+        # breakpoint()
+
+        if IDexterityContent.providedBy(self.context):
+            context = self.context
+        else:
+            context = self.context.aq_parent
+        if not hasattr(aq_base(context), 'query'):
```

Comment:
This validator is a nice addition, but currently it looks like as if it was only testing for the context being a collection and would break if you add a target_collection reference to somewhere else.

Reply:
That's true ... @agitator we need a bit more logic to validate referenced collection UIDs